metrics-exporter/src/bin/io_engine/cache/mod.rs

```diff
@@ -58,21 +62,13 @@ impl Data {
     fn new() -> Self {
         Self {
             pools: Pools { pools: vec![] },
+            pool_stats: PoolIoStats { pool_stats: vec![] },
         }
     }
 }
 
-/// To store data in shared variable i.e cache.
-pub(crate) async fn store_data(client: GrpcClient) {
-    tokio::spawn(async move {
-        store_resource_data(client).await;
-    });
-}
-
-/// To store pools related data in cache.
-async fn store_resource_data(client: GrpcClient) {
-    loop {
-        let _ = pool::store_pool_info_data(client.clone()).await;
-        sleep(ExporterConfig::get_config().polling_time()).await;
-    }
+/// Populates Resource cache struct.
+pub(crate) async fn store_resource_data(client: &GrpcClient) {
+    let _ = pool::store_pool_info_data(client.clone()).await;
```

Comment:
why is one of them receiving by value and the other by reference?

Reply:
both receive reference now